ignite/metrics/accuracy.py

```diff
@@ -161,11 +161,11 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
             y = torch.transpose(y, 1, last_dim - 1).reshape(-1, num_classes)
             correct = torch.all(y == y_pred.type_as(y), dim=-1)
 
-        self._num_correct += torch.sum(correct).item()
+        self._num_correct += torch.sum(correct).to(self._device)
```

Comment:
Maybe, it would be better to put a comment here why we do not call detach on `correct`...